README.md

```diff
@@ -2217,4 +2217,219 @@ A Japanese word meaning "form", relating to karate. Used commonly in programming
+</body>
+</html>
+```
+
+#### HTML Symbols
+
+| Symbol | Definition |
+| ------ | ---------- |
+| `<p>`  | paragraph  |
+| `<b>`  |   body     |
+| `<div>` | defines a division or section |
+| `<!-- -->` | comments in the middle of symbol |
+
+#### HTML vs Markdown 
+
+| Markdown | HTML |
+| -------- | ---- |
+|  `#`  | `<h1>` |
+|  `-`  | `<ul> <li> text goes here </li> </ul>` |
+|  `1.` | `<ol> <li> text goes here </li> </ol>` |
+| backtick | `<pre>` |
+| ` ```java | `<pre><code></code></pre>` |
+| `**` | `<b>` |
+
+## Functional Programming with JavaScript
+
+Functional Programming is the combination of declarative coding (telling code what you want it to do) and immutable (unchangeable) data.
+
+JavaScript allows programmers to put the output of one function into another function.
+
+In the example below, values are assigned to the functions `a`, `b`, and `c` and these functions are then put into the function `i`.
+
+``` javascript
+let a = undefined;
+let b = null;
+let c = "";
+let d = new Date();
+let e = 1;
+let f = false;
+let g = ["g",1];
+let h = {h: "foo"};
+let i = function(a,b,c) {
+    console.log(a,b,c);
+}
+```
+
+An example of a variable within a function:
+
+```javascript
+let j = function(myVar) {
+    myVar("hello")
+}
+```
+
+In JavaScript, for loops become `forEach` and include a variable name.
+
+The following example assigns 5 numbers to an array, then filters that array to return only even numbers. Then the even numbers are assigned to `foo` and printed. Lastly, the entire array is printed.
+
+```javascript
+let array = [1,2,4,6,7]
+
+array.filter(function(n){ return n % 2 === 0}) // classic function
+    .map(n => "foo: " + n) // arrow function
+    .forEach(n => console.log(n))
+
+console.log(array);
+```
+
+It's important to note that you should never change the nature of an array once it's been defined (e.g. once `array =` you should never write `array =` again in that code).
+
+The example below shows the classic FizzBuzz in JavaScript.
+
+```javascript
+for(let i = 1; i <= 100; i++){
+    if(i % 15 === 0) {
+        console.log("FizzBuzz");   
+    } else if(i % 3 === 0) {
+        console.log("Fizz");  
+    } else if(i % 5 === 0) {
+        console.log("Buzz");
+    } else {
+        console.log(i);
+    }
+}
+```
+
+This could also be done as using an array. You can add to arrays using `push`.
+
+```javascript
+let a = []
+for(let i = 1; i <= 100; i++){
+    a.push(i)
+}
+
+a.map(i => {
+    if (i % 15 === 0) {
+        return "FizzBuzz"
+    }
+    if (i % 5 === 0) {
+        return "Buzz"
+    }
+    if (i % 3 === 0) { 
+        return "Fizz"
+    }
+    return i
+}).forEach(i => console.log(i))
+```
+
+## Computer Programming
+
+Order = the rate of growth of a function.
+
+| Order | Time |
+| ----- | ---- |
+| O(1) | fast |
+| O(log n) |  `|` |
+| O(nlog n) | `|`  |
+| O(n) |  `|` |
+| O(n<sup>2</sup>) | `|`  |
+| O(2<sup>n</sup>) | `v`  |
+| O(n!) | slow |
+
+`O(n)` is used for searching unsorted lists.
```

Comment:
Saying that it's "used for" is not quite right IMO. It makes it sound like we can choose any of these always, which isn't the case. It's more that `O(n)` is the best we can do in this scenario.